CRM/Pledge/Form/Search.php

```diff
@@ -98,6 +98,11 @@ public function preProcess() {
     }
 
     if ($this->_force) {
+      // pledge related dates
+      $this->addSearchFieldMetadata(['Pledge' => CRM_Pledge_BAO_Query::getSearchFieldMetadata()]);
+      $this->addSearchFieldMetadata(['PledgePayment' => CRM_Pledge_BAO_Query::getPledgePaymentSearchFieldMetadata()]);
+      $this->addFormFieldsFromMetadata();
+      $this->setFormValues();
```

Comment:
So this is a change from the approach elsewhere - if I look at CRM_Contribute_Form_Search we are relying on this being done in parent::setDefaults (which would run here too) or in preProcess - which is not very 'mature' in that the forms are not sharing a function.

I'm gonna poke around a bit more on this one

Reply:
I couldn't see how parent::setDefaults was setting the formValues where as setFormValues was calling setDefaults() and this is in the situation where we have &force=1 in the URL param

Reply:
ok i think re-looking at the Contribute search this function is actually called in the postProcess function so i'll move it there